### PR TITLE
Added CosmosDB Configuration and support for --secrets-volume flag

### PR DESCRIPTION
**Pull request by gortegaMS**
- Added CosmosDB classes to create connection
- Added support for --secrets-volume flag and secrets validation
- Created initial README.md file

Closes [#Issue 279](https://github.com/retaildevcrews/wcnp/issues/279)

### CODE REVIEW

File: src/DataAccessLayer/CosmosConfig.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Microsoft.Azure.Cosmos;
+
+namespace Ngsa.LodeRunner.DataAccessLayer
```
Comment (joheec) [inline src/DataAccessLayer/CosmosConfig.cs:7]: Ngsa?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
+# NGSA Loderunner - A web request validation tool
+
+Loderunner (L8s) is an internal web request validation tool that we use to run end-to-end tests and long-running smoke tests.
```
Comment (joheec) [inline README.md:3]: L8s? but "oderunner" has 9 characters.
I feel like L8r would make more sense.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+# pull image from GitHub Container Repository 
+docker pull ghcr.io/retaildevcrews/ngsa-lr:beta
+
+```
+
+Experiment with L8s
+
+```bash
+# run L8s with --help option; this should output command line options shown below
```
Comment (joheec) [inline README.md:21]: You could also add additional markdown to direct someone to "below"

I think it's something like
[below](#command-line-parameter)

File: README.md (hunk 2 above)
Comment (gortegaMS) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (gortegaMS) [reply README.md]: Fixed

File: README.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  - -h
+  - other parameters are ignored
+  - environment variables are ignored
+- --dry-run bool
+  - -d
+    - validate parameters but do not execute tests
+- --server string1 [string2 string3]
+  - -s
+  - SERVER
+    - server Url (i.e. `https://ngsa-central-asb-test.cse.ms`)
```
Comment (jofultz) [inline README.md:81]: Let's remove the actual Url and put a fictional example.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+Loderunner works in two distinct modes. The default mode processes the input file(s) in sequential order one time and exits. The "run loop" mode runs in a continuous loop until stopped or for the specified duration. Some environment variables and command flags are only valid if run loop is specified and L8r will exit and display usage information. Some parameters have different default values depending on the mode of execution.
+
+### Example Arguments for Long Running Tests
+
+```bash
+# continuously send request every 15 seconds
+
+docker run --rm ghcr.io/retaildevcrews/ngsa-lr:beta --sleep 15000 --run-loop --server "https://DomainName" --files benchmark.json
+```
```
Comment (jofultz) [inline README.md:46]: Since this is new doc for Loderunner, have all of the docker run commands been tested?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    - default location current directory
+    - `required`
+- --base-url string
+  - -u
+  - BASE_URL
+    - base URL and optional path to the test files (http or https)
+- --delay-start int
+  - DELAY_START
+    - delay starting the validation test for int seconds
+    - default `0`
```
Comment (jofultz) [inline README.md:96]: Need to document "-1" usage and requirements.

Comment (jofultz) [inline README.md:96]: Need to document "-1" usage.

**Review by jofultz — APPROVED** [review]
I have a couple of readme changes.  Other than that, it's a good start.  Next week we'll start looking at refactoring the data lib out of Loderunner into a separate lib.

File: README.md (hunk 5 above)
Comment (gortegaMS) [reply README.md:46]: yes I did, and also removed documentation for args that are not implemented for loderunner

File: README.md (hunk 4 above)
Comment (gortegaMS) [reply README.md]: Fixed

File: README.md (hunk 6 above)
Comment (gortegaMS) [reply README.md:99]: Added documentation for  --delay-start "-1 " and "--secrets-volume"

**Pull request merged**